src/bootstrap/install.rs

```diff
@@ -251,6 +251,11 @@ install!((self, builder, _config),
     };
 );
 
+fn should_build_llvm(config: &Config) -> bool {
+    // additionaly check that llvm is not externally provided
```

Comment:
Can you write this as a doc-comment on top of this function? It will be better for other devs to directly read this via LSP/rust-analyzer.